packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -363,6 +411,113 @@ contract OPContractsManager is ISemver {
         return output;
     }
 
+    /// @notice addGameType deploys a new dispute game and links it to the DisputeGameFactory. The inputted _gameConfigs
+    /// must be added in ascending GameType order.
+    function addGameType(AddGameInput[] memory _gameConfigs) external returns (AddGameOutput[] memory) {
+        if (address(this) == address(thisOPCM)) revert OnlyDelegatecall();
+        if (_gameConfigs.length == 0) revert InvalidGameConfigs();
+
+        AddGameOutput[] memory outputs = new AddGameOutput[](_gameConfigs.length);
+        Blueprints memory bps = thisOPCM.blueprints();
+
+        // Store last game config to as an int64 so that we can ensure that the same game config is not added twice.
+        // An int64 is OK to use here since the GameType is a uint32. We start at -1 since 0 is a valid GameType.
+        int64 lastGameConfig = -1;
+
+        for (uint256 i = 0; i < _gameConfigs.length; i++) {
+            AddGameInput memory gameConfig = _gameConfigs[i];
+
+            // This conversion is safe because the GameType is a uint32, which will always fit in an int64.
+            int64 gameTypeInt = int64(uint64(gameConfig.disputeGameType.raw()));
+            // Ensure that the game configs are added in ascending order, and not duplicated.
+            if (lastGameConfig >= gameTypeInt) revert InvalidGameConfigs();
+            lastGameConfig = gameTypeInt;
+
+            // Grab the FDG from the SystemConfig.
+            IFaultDisputeGame fdg = IFaultDisputeGame(
+                address(
+                    IDisputeGameFactory(gameConfig.systemConfig.disputeGameFactory()).gameImpls(
+                        GameTypes.PERMISSIONED_CANNON
+                    )
+                )
+            );
+            // Pull out the chain ID.
+            uint256 l2ChainId = fdg.l2ChainId();
+
+            // Deploy a new DelayedWETH proxy for this game.
+            outputs[i].delayedWETH = IDelayedWETH(
+                payable(deployProxy(l2ChainId, gameConfig.proxyAdmin, gameConfig.saltMixer, "DelayedWETH"))
+            );
+
+            // Initialize the proxy.
+            upgradeAndCall(
+                gameConfig.proxyAdmin,
+                address(outputs[i].delayedWETH),
+                thisOPCM.implementations().delayedWETHImpl,
```

Comment:
nit, small optimization to cache this address outside the loop

Reply:
I can try but I was running into stack too deep errors.

Reply:
yeah, pulling this out leads to stack too deep. if it's a big deal i can see about refactoring the method more to avoid this, but it seems clear enough as-is.